mantis-publish/mantis-publish-core/build.gradle

```diff
@@ -22,6 +22,7 @@ ext {
 
 dependencies {
     implementation "io.mantisrx:mql-jvm:$mqlVersion"
+    implementation project(":mantis-common")
```

Comment:
[For ur consideration] mantis-common probably will bring in quite some unused deps to the publish side of things which becomes part of the client MRE deps. I feel that it would be nice if the serde part can be separated out in this case.

Reply:
@Andyz26 K I will try making `mantis-serde` which will depend on `mantisShaded`. `mantisShaded` has some extra stuff in it, but that's already being pulled into the client through `mantis-discovery-proto`. I'll open a seperate PR for seperating out the serde and get that other PR merged before this one.

Reply:
Updated this pr to use `mantis-common-serde`